packages/foam-vscode/src/features/preview/wikilink-embed.ts

```diff
@@ -177,7 +177,11 @@ function fullExtractor(
   parser: ResourceParser,
   workspace: FoamWorkspace
 ): string {
-  let noteText = readFileSync(note.uri.toFsPath()).toString();
+  let noteText;
+
+  vsWorkspace.fs
```

Comment:
Are you sure this will work?
I am not sure this would work, as this call returns a promise, and whether `noteText` is actually set when needed (e.g. line `192`) becomes a matter of timing.
This is the reason why originally we didn't migrate this part of the code from `readFileSync`.

Am I missing something?

Reply:
It is not working. Need to refine it, but needed a test run on thr CI as I had some issues locally. Forgot to add wip to the PR ;)